spec/models/manageiq/providers/cloud_manager/auth_key_pair_spec.rb

```diff
@@ -1,20 +1,26 @@
 RSpec.describe ManageIQ::Providers::CloudManager::AuthKeyPair do
-  let(:ems) { FactoryBot.create(:ems_cloud) }
+  let(:ems) { ManageIQ::Providers::CloudManager.new }
```

Comment:
Is this just so you can test that the base-class doesn't support by default down [[here]](https://github.com/ManageIQ/manageiq/pull/21457/files#diff-f0a3a431bbdf907ee3825f56251e74820f842511b04bf4f5e2513dffeedabbcdR75) ?  Think it'd be better to drop that one test rather than this and have to stub `:id` and `:hostname_format_valid?` everywhere

Reply:
I agree.

Keep the `let` here the way it was
And either put a `let` into the validation context down below, or just add an `ems = ...` into the `it` block down below.

Reply:
@agrare what bothers me is that `FactoryBot.create(:ems_cloud)` resolves to an AWS provider which may have its own behavior.  I think it makes more sense to change `FactoryBot.create(:ems_cloud)` to resolve to a dummy_cloud_provider that is only implemented in specs.  But probably that should be done in a future PR focused on that refactoring.

Reply:
Yes completely agree, that's been an issue ever since we added validation that only leaf EMS classes can be created.  @Fryguy and I had discussed creating spec-only leaf EMS classes so that the "base" factories aren't any specific provider plugin.

Reply:
maybe we should add dummy leaf EMS classes in the core repo that we can use for specs - but somehow disable them in production.

Reply:
I also ran into this today being frustrated that I was creating an object that was not in the core.

I wonder if it makes sense to introduce `manageiq-providers-dummy` that is only included in tests.
it would have an infra and a cloud component. Heck, it would have one of everything.

Think we would use `expects` to handle `supports` though